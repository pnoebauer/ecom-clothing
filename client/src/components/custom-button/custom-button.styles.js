import styled, {css} from 'styled-components';

const buttonStyles = css`
	background-color: black;
	color: white;
	border: none;

	&:hover {
		background-color: white;
		color: black;
		border: 1px solid black;
	}
`;

const invertedButtonStyles = css`
	background-color: white;
	border: 1px solid black;
	box-shadow: inset 0 5px 2px -9px rgba(0, 0, 0, 0.4),
		0 3px 6px 0 rgba(255, 255, 255, 0.3);
	color: black;
	text-shadow: 0 2px 5px white;

	&:hover {
		background-color: black;
		color: white;
		border: none;
		box-shadow: inset 0 5px 2px -9px rgba(255, 255, 255, 0.2),
			0 3px 6px 0 rgba(0, 0, 0, 0.1);
		text-shadow: 0 2px 5px black;
	}

	&:active {
		box-shadow: inset 0 2px 2px -9px rgba(255, 255, 255, 0.2),
			0 1px 6px 0 rgba(0, 0, 0, 0.1);
	}
`;

const googleSignInStyles = css`
	background-color: #4285f4;
	color: white;
	border: none;

	&:hover {
		background-color: #357ae8;
		border: none;
	}
`;

const getButtonStyles = props => {
	if (props.isGoogleSignIn) {
		return googleSignInStyles;
	}

	return props.inverted ? invertedButtonStyles : buttonStyles;
};

export const CustomButtonContainer = styled.button`
	min-width: 165px;
	width: auto;
	height: 50px;
	letter-spacing: 0.5px;
	line-height: 50px;
	padding: 0 35px 0 35px;
	font-size: 12px;
	text-transform: uppercase;
	font-family: 'Open Sans Condensed';
	font-weight: bolder;
	cursor: pointer;
	outline: none;
	border-radius: 5px;
	display: flex;
	justify-content: center;
	${getButtonStyles};

	&:active {
		transform: scale(0.98);
	}
`;
