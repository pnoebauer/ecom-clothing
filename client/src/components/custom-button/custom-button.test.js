import React from 'react';
import {shallow} from 'enzyme';

import CustomButton from './custom-button.component';
import {CustomButtonContainer} from './custom-button.styles';

describe('CustomButton component tests', () => {
	let wrapper;
	let mockClick;

	const mockChildren = 'Sample Input';
	const mockClassName = 'test';

	beforeEach(() => {
		mockClick = jest.fn();

		const mockProps = {
			className: mockClassName,
			onClick: mockClick,
		};

		wrapper = shallow(<CustomButton {...mockProps}>{mockChildren}</CustomButton>);
	});

	test('render CustomButton component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('call mockAddItem when the AddButton is clicked', () => {
		wrapper.find(`.${mockClassName}`).simulate('click');
		expect(mockClick).toHaveBeenCalled();
	});
});
