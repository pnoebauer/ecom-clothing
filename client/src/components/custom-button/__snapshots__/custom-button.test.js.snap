// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton component tests render CustomButton component 1`] = `
<styled.button
  className="test"
  onClick={[MockFunction]}
>
  Sample Input
</styled.button>
`;
